Update Sync Tab UI to Include Pressable Sites

epeicher opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "Fixes" keyword and use "Related to" instead.
-->

## Related issues

- Resolves STU-349

## Proposed Changes

- Updated sync tab UI text to explicitly mention Pressable sites

Figma link RToz6tIuQ7nlZrikBte4GU-fi-7835_3430

|Before | After|
|-------|------|
|  ![CleanShot 2025-04-23 at 11 56 43@2x](https://github.com/user-attachments/assets/5d21c0a8-ae1b-4580-934b-777843694d43)|    ![CleanShot 2025-04-23 at 11 57 44@2x](https://github.com/user-attachments/assets/a1f02d01-44ab-4490-a2b7-149592fd523f)|

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

- Apply this branch
- Run `STUDIO_PRESSABLE_SYNC=true npm start` 
- Using a site without _synced_ sites, check that the text in the `Sync` tab corresponds with the Figma designs. Link here RToz6tIuQ7nlZrikBte4GU-fi-7835_3430
- Check that the text looks fine for different sizes of the screen
- Run Studio again without the feature flag: `npm start`
- Using a site without _synced_ sites, check that the text in the `Sync` tab corresponds to the `Before` screenshot

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [x] Have you checked for TypeScript, React or other console errors?

epeicher committed:
Sync: Change copy to include Pressable

ivan-ottinger reviewed (commented):
The change looks good. 👌🏼 

We just may want to hide it behind the feature flag. → To ensure the tab does not show the updated copy until we are launching the Pressable sync.

I have also one thought regarding the updated copy, but I will bring this up in the Linear issue so we can discuss further with Marina: STU-320-linear-issue#comment-3f6020e7.

epeicher commented:
> We just may want to hide it behind the feature flag. → To ensure the tab does not show the updated copy until we are launching the Pressable sync.

Good point @ivan-ottinger ! I will use the feature flag for this

sejas reviewed (commented):
As @ivan-ottinger already mentioned. Let's add any change behind the feature flag.
Similar to: https://github.com/Automattic/studio/blob/d85c3c48e738ee2310fd21e03e4fec6ced2113e8/src/hooks/use-fetch-wpcom-sites/index.tsx#L149

epeicher committed:
Update copy and use the feature flag to display the new copy

epeicher commented:
@ivan-ottinger, @sejas, as suggested, I have updated the copy and added the feature flag in this commit 39fa121. I have also updated the test steps accordingly. This is ready for another review.

sejas reviewed (approved):
Looks good to me. Thanks for adding the change in the text behind a feature flag.
We can revisit the copy changes later in the project.

katinthehatsite reviewed (approved):
The changes look good to me and the description looks clear as well!
